plugins/redmine.py

```diff
@@ -0,0 +1,156 @@
+from redminelib import Redmine
+from utils.utils import echo, echo_error, echo_success, echo_skip
+
+
+class RedminePlugin(object):
+    def __init__(
+        self,
+        redmine_url,
+        redmine_username,
+        redmine_password,
+        redmine_task_prefix="redmine",
+        sync=False,
+        jira=None
+    ):
+        self.redmine = Redmine(
+            redmine_url,
+            username=redmine_username,
+            password=redmine_password
+        )
+        self.redmine_task_prefix = redmine_task_prefix
+        self.sync = sync
+        self.jira = jira
+
+    def get_issues(self, redmine_userid):
+        user = self.redmine.user.get(redmine_userid)
+        return list(
+            # use (status_id='open|closed') to filter out issues
+            self.redmine.issue.filter(assigned_to_id=user.id, status_id='*')
+        )
+
+    def do_sync(self, issue, jira_username):
+        if not self.jira:
+            echo_error("Jira Wrapper is not Available")
+            return
+        # build the unique issue_text prefix
+        issue_text = '{prefix}#{issue.project}#{issue.id}#'.format(
+            prefix=self.redmine_task_prefix,
+            issue=issue
+        )
+
+        # check if it already exists
+        search_query = (
+            'project = {} '
+            'AND status != Done '
+            'AND assignee = {} '
+            'AND summary ~ \\"{}\\"'
+        ).format(
+            self.jira.project_id,
+            jira_username,
+            issue_text.replace('#', '\u0023')
+        )
+        echo(
+            "Searching Jira for {0} using query [{1}]".format(
+                issue_text, search_query
+            )
```

Comment:
do you think we need to add the username as well while building search string query?

Reply:
I tested some searches and the only one that worked for the workflow was this one including the assignee.

I think we can share the same query between plugins.

https://github.com/rochacbruno/jirasync/blob/redmine/plugins/redmine.py#L46-L50

Reply:
As discussed with you we can add that in some common location soon. This PR looks really good as first running version 1.0